#2414: parse SSH repositories url with a commit hash

GMLudo opened the pull request:
When you have a VCS repository with SSH, and you want to clone a specific commit, like you have in `requirements.txt` (ex: `git@github.com:Eyepea/pip.git@8cf54fff31b650847e0cddc2cd2951c34e0b4822`) you have a git command error, as described in #2414 issue.
This commit fixes that.

msabramo commented:
It looks like this change had 2 small flake8 problems that need to be corrected:

~~~
pep8 runtests: commands[0] | flake8 .
./pip/vcs/__init__.py:156:80: E501 line too long (86 > 79 characters)
./pip/vcs/__init__.py:157:1: W293 blank line contains whitespace
ERROR: InvocationError: '/home/travis/build/pypa/pip/.tox/pep8/bin/flake8 .'
~~~

xavfernandez commented:
Thanks for the fix but some tests for `get_url_rev` (at least for your use case) would also be nice.

GMLudo commented:
@msabramo: pep8 issues should be fixed. Sorry for that, at the company where I work, the standard is 160 characters max.
@xavfernandez: I've added tests for my use cases and I've added another fix.

Do you need something else before the merge ?

msabramo commented:
Still has `pep8` failures:

```
pep8 runtests: commands[0] | flake8 .
./tests/unit/test_vcs.py:49:80: E501 line too long (100 > 79 characters)
./tests/unit/test_vcs.py:52:80: E501 line too long (96 > 79 characters)
./tests/unit/test_vcs.py:56:80: E501 line too long (102 > 79 characters)
```

You can run `tox -e pep8` to check.

GMLudo commented:
Thanks a lot for your PR on my PR, I've merged.

Now, I think it's the right time to merge this PR, isn't it ? ;-)

GMLudo commented:
@dstufft : thanks for the merge

dstufft commented:
This was reverted due to breaking things, I've opened a new PR with the changes at https://github.com/pypa/pip/pull/2551 so we don't lose this PR which also includes what broke.
